### PR TITLE
fix: about event view on collapsing layout

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Load event image into the view
- Set title bar to event name or "" when appBarLayout is collapsed or expanded

Fixes: #1488

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 1]
```diff
@@ -12,24 +12,24 @@ import androidx.lifecycle.Observer
...
 import org.fossasia.openevent.general.event.EventUtils
 import org.fossasia.openevent.general.utils.extensions.nonNull
 import org.fossasia.openevent.general.utils.stripHtml
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
+class AboutEventFragment : Fragment() {
     private lateinit var rootView: View
     private val aboutEventViewModel by viewModel<AboutEventViewModel>()
-    private var isHideToolbarView: Boolean = false
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: Consider renaming this to `hideToolbarView` or `showToolbarView`. `isHideToolbarView` does not seem appropriate in my opinion

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 2]
```diff
@@ -75,26 +73,23 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
         rootView.aboutEventContent.text = event.description?.stripHtml()
         val startsAt = EventUtils.getEventDateTime(event.startsAt, event.timezone)
         val endsAt = EventUtils.getEventDateTime(event.endsAt, event.timezone)
+
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: Unnecessary change

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 3]
```diff
@@ -75,26 +73,23 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
         rootView.aboutEventDetails.text = EventUtils.getFormattedDateTimeRangeBulleted(startsAt, endsAt)
         rootView.eventName.text = event.name
-    }
-
-    override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {
-        val maxScroll = appBarLayout.totalScrollRange
-        val percentage = Math.abs(verticalOffset).toFloat() / maxScroll.toFloat()
+        Picasso.get()
+            .load(event.originalImageUrl)
+            .into(rootView.aboutEventImage)
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: I think adding a placeholder might be a good idea

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 4]
```diff
@@ -75,26 +73,23 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
-        if (percentage < 1f && !isHideToolbarView) {
-            // Not Collapsed
-            rootView.detailsHeader.visibility = View.VISIBLE
-            rootView.aboutEventCollapsingLayout.title = " "
-            isHideToolbarView = !isHideToolbarView
-        }
+        rootView.appBar.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { appBarLayout, offset ->
+            val thisActivity = activity
+            if (Math.abs(offset) == appBarLayout.getTotalScrollRange()) {
+                rootView.detailsHeader.visibility = View.GONE
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: You can change this to ` rootView.detailsHeader.isVisible = true` .

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt (hunk 1 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: This red line means I deleted this because it is not necessary

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt (hunk 3 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: yeah, it's a good idea, I will add it

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt (hunk 4 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: I think it is the same right?

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt (hunk 1 above)
Comment (addiegupta) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: My bad sorry about that

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt (hunk 4 above)
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: No, use .isVisible, as it's in accordance with kotlin code style. What you are using now is in accordance with java code style.

**Comment by anhanh11001** [comment]
@liveHarshit please review

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -74,8 +74,8 @@
     <dimen name="event_details_divider">1.2dp</dimen>
     <dimen name="divider_margin_bottom">20dp</dimen>
     <dimen name="divider_margin_top">12dp</dimen>
-    <dimen name="collapsing_toolbar_height">220dp</dimen>
-    <dimen name="details_header_margin_top">40dp</dimen>
+    <dimen name="collapsing_toolbar_height">320dp</dimen>
+    <dimen name="details_header_margin_top">100dp</dimen>
```
Comment (liveHarshit) [inline app/src/main/res/values/dimens.xml]: Add dimensions in dimension resources file.

Comment (anhanh11001) [reply app/src/main/res/values/dimens.xml]: @liveHarshit this is dimensions inside dimens.xml file... Can you please clarify your review?

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 6]
```diff
@@ -75,26 +73,24 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
-            isHideToolbarView = !isHideToolbarView
-        }
+        rootView.appBar.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { appBarLayout, offset ->
+            val thisActivity = activity
+            if (Math.abs(offset) == appBarLayout.getTotalScrollRange()) {
+                rootView.detailsHeader.isVisible = false
+                if (thisActivity is AppCompatActivity) thisActivity.supportActionBar?.title = event.name
+            } else {
+                rootView.detailsHeader.isVisible = true
+                if (thisActivity is AppCompatActivity) thisActivity.supportActionBar?.title = ""
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: Just cast one time, when you define the value for `thisActivity`.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: Updates?

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: oh yeah, I forgot to do that, fixing now

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: @liveHarshit I have fixed it

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 7]
```diff
@@ -67,30 +65,27 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
-            isHideToolbarView = !isHideToolbarView
-        }
-        if (percentage < 1f && !isHideToolbarView) {
-            // Not Collapsed
-            rootView.detailsHeader.visibility = View.VISIBLE
-            rootView.aboutEventCollapsingLayout.title = " "
-            isHideToolbarView = !isHideToolbarView
-        }
+        rootView.appBar.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { appBarLayout, offset ->
+            val thisActivity = activity as AppCompatActivity
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: Do not cast, use smart cast instead.

**Comment by angmas1** [comment]
Use setToolbar method from utils to setToolbar

**Comment by anhanh11001** [comment]
@angmas1 actually, I do import setToolbar from Utils file and do use it. Which part of the PR are you mentioning to? Thanks!

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 8]
```diff
@@ -67,30 +65,29 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
-            // Not Collapsed
-            rootView.detailsHeader.visibility = View.VISIBLE
-            rootView.aboutEventCollapsingLayout.title = " "
-            isHideToolbarView = !isHideToolbarView
-        }
+        rootView.appBar.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { appBarLayout, offset ->
+            val thisActivity = activity
+            if (thisActivity is AppCompatActivity) {
+                if (Math.abs(offset) == appBarLayout.getTotalScrollRange()) {
+                    rootView.detailsHeader.isVisible = false
```
Comment (angmas1) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: I was referring to this part. Never mind, I will send a PR to fix this.

File: app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt [hunk 9]
```diff
@@ -67,30 +65,29 @@ class AboutEventFragment : Fragment(), AppBarLayout.OnOffsetChangedListener {
...
+                    thisActivity.supportActionBar?.title = event.name
```
Comment (angmas1) [inline app/src/main/java/org/fossasia/openevent/general/about/AboutEventFragment.kt]: I was referring to this part. Never mind, I will send a PR to fix this.
